react/features/app/actions.js

```diff
@@ -32,6 +34,8 @@ declare var APP: Object;
  */
 export function appNavigate(uri: ?string) {
     return async (dispatch: Dispatch<any>, getState: Function) => {
+        dispatch(endAllSessions());
```

Comment:
Right. I'm not sure if I like it either. But what else could be done ? Should we start dispatching `APP_NAVIGATE` action ? The maybe middleware in base/session would figure out how many sessions can stay alive ?

Reply:
> The maybe middleware in base/session would figure out how many sessions can stay alive ?

I think so. If a createSession action is dispatched but there is another session in progress, we terminate the current one, for instance.

Reply:
>I think so. If a createSession action is dispatched but there is another session in progress, we terminate the current one, for instance.

It is not enough to end the session on `createSession` action, because we need to handle the case of hangup => `appNavigate(undefined)` where no session is created.